aten/src/ATen/native/native_functions.yaml

```diff
@@ -3711,10 +3727,18 @@
 # Return: (Tensor output, Tensor indices)
 - func: max_pool2d_with_indices(Tensor self, int[2] kernel_size, int[2] stride=[], int[2] padding=0, int[2] dilation=1, bool ceil_mode=False, *, Tensor(a!) output, Tensor(b!) indices) -> (Tensor(a!), Tensor(b!))
   python_module: nn
+  dispatch:
+    CPU: max_pool2d_with_indices_out
+    CUDA: max_pool2d_with_indices_out
+    MkldnnCPU: mkldnn_max_pool2d_with_indices_out
 
 # Return: (Tensor output, Tensor indices)
 - func: max_pool2d_with_indices(Tensor self, int[2] kernel_size, int[2] stride=[], int[2] padding=0, int[2] dilation=1, bool ceil_mode=False) -> (Tensor, Tensor)
```

Comment:
BTW: There is also `max_pool2d` that doesn't need to return indices, useful for inference workload. We can implement it more efficiently with MKL-DNN by setting the `prop_kind` to `forward_inference`.